File: Week 1/insert.hs
insert :: [Int] -> Int -> [Int]
insert [] e
    = [e]
insert (x:xs) e
    | e < x = e:x:xs
    | otherwise = x:insert xs e

main :: IO ()
main
    = putStrLn(show(insert [1,2,3] 4))

File: Week_10/insert.pl
insert( E, [], [E] ).
insert( E, [H|T], [E,H|T] )
    :- E < H.
insert( E, [H|T], [H|W] )
    :- E >= H,
        insert( E, T, W ).

main
    :- insert( 6, [ 1, 4, 5, 7, 9, 12 ], R ),
        write( R ).